src/concrete/ml/sklearn/base.py

```diff
@@ -1281,19 +1290,65 @@ def __init_subclass__(cls):
                 _TREE_MODELS.add(cls)
                 _ALL_SKLEARN_MODELS.add(cls)
 
-    def __init__(self, n_bits: int):
+    def __init__(self, n_bits: Union[int, Dict[str, int]]):
         """Initialize the TreeBasedEstimatorMixin.
 
         Args:
-            n_bits (int): The number of bits used for quantization.
+            n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and leaves. If a dict is
+                passed, then it should contain "op_inputs" and "op_leaves" as keys with
+                corresponding number of quantization bits so that:
+                    - op_inputs (mandatory): number of bits to quantize the input values
+                    - op_leaves (optional): number of bits to quantize the leaves
+                Default to 6.
         """
-        self.n_bits: int = n_bits
+
+        # Check if 'n_bits' is a valid value.
+        _inspect_tree_n_bits(n_bits)
+
+        #: The number of bits to quantize the model.
+        self.n_bits: Union[int, Dict[str, int]] = n_bits
 
         #: The model's inference function. Is None if the model is not fitted.
         self._tree_inference: Optional[Callable] = None
 
+        #: Wether to perform the sum of the output's tree ensembles in FHE or not.
+        self._use_fhe_sum = False
+
         BaseEstimator.__init__(self)
 
+    @property
+    def use_fhe_sum(self) -> bool:
+        """Property getter for `use_fhe_sum`.
+
+        Returns:
+            bool: The current setting of the `_use_fhe_sum` attribute.
+        """
+        return self._use_fhe_sum
+
+    @use_fhe_sum.setter
+    def use_fhe_sum(self, value) -> None:
+        """Property setter for `use_fhe_sum`.
+
+        Args:
+            value (int): Whether to enable or disable the feature.
+        """
+
+        assert isinstance(value, bool), "Value must be a boolean type"
+
+        if value is True:
+            warnings.simplefilter("always")
+            warnings.warn(
+                "Enabling `use_fhe_sum` computes the sum of the ouputs of tree ensembles in FHE.\n"
+                "This may slow down the computation and increase the maximum bitwidth.\n"
+                "To optimize performance, consider reducing the quantization leaf precision.\n"
```

Comment:
Can they really decrease the output quantization precision? I don't think we have made this feature available

Reply:
> I don't think we have made this feature available

Yes, I added it in this PR, see: src/concrete/ml/quantization/post_training.py::_get_n_bits_dict_trees

Reply:
oh awesome! thanks